File: Embassat/View/Tickets/CADEMTicketsViewController.m
//
//  CADEMTicketsViewController.m
//  Embassat
//
//  Created by Joan Romano on 22/05/14.
//  Copyright (c) 2014 Crows And Dogs. All rights reserved.
//

#import "CADEMTicketsViewController.h"

@interface CADEMTicketsViewController ()

@property (nonatomic, strong) IBOutletCollection(UILabel) NSArray *titleLabels;
@property (nonatomic, strong) IBOutletCollection(UILabel) NSArray *bodyLabels;

@property (nonatomic, weak) IBOutlet UIView *seasonContainer, *dayTicketContainer, *hotelContainer, *tresCContainer;

@end

@implementation CADEMTicketsViewController

- (instancetype)init
{
    if (self = [super init])
    {
        self.title = @"ENTRADES";
    }
    
    return self;
}

- (void)viewDidLoad
{
    [super viewDidLoad];
    
    [@[self.seasonContainer, self.dayTicketContainer, self.hotelContainer, self.tresCContainer] enumerateObjectsUsingBlock:^(UIView *subview, NSUInteger idx, BOOL *stop) {
        UITapGestureRecognizer *tapGesture = [[UITapGestureRecognizer alloc] init];
        [tapGesture.rac_gestureSignal subscribeNext:^(id x) {
            [self linkPressed:subview];
        }];
        
        [subview addGestureRecognizer:tapGesture];
    }];
    
    [self.titleLabels enumerateObjectsUsingBlock:^(UILabel *label, NSUInteger idx, BOOL *stop) {
        label.font = [UIFont em_boldFontOfSize:16.0];
    }];
    [self.bodyLabels enumerateObjectsUsingBlock:^(UILabel *label, NSUInteger idx, BOOL *stop) {
        label.font = [UIFont em_romanFontOfSize:16.0];
    }];
}

#pragma mark - Actions

- (IBAction)linkPressed:(UIView *)sender
{
    [self openTicketLink:[NSURL URLWithString:sender.tag == 3 ? @"http://www.tresc.cat/fitxa/concerts/43054/Embassat-2014" : @"https://www.ticketea.com/embassat-2014-festival-independent-del-valles/"]];
}

- (void)openTicketLink:(NSURL *)link
{
    [[UIApplication sharedApplication] openURL:link];
}

@end

File: Embassat/View/Tickets/CADEMTicketsViewController.swift
//
//  CADEMTicketsViewController.swift
//  Embassa't
//
//  Created by Joan Romano on 14/05/15.
//  Copyright (c) 2015 Crows And Dogs. All rights reserved.
//

import UIKit

public class CADEMTicketsViewController: CADEMRootViewController {
    
    @IBOutlet var titleLabels: [UILabel]?
    @IBOutlet var bodyLabels: [UILabel]?
    @IBOutlet weak var seasonContainer: UIView?
    @IBOutlet weak var dayTicketContainer: UIView?
    @IBOutlet weak var petitEmTicketContainer: UIView?
    
    override public func viewDidLoad() {
        super.viewDidLoad()
        
        title = "Entrades"
        
        for view in [seasonContainer, dayTicketContainer, petitEmTicketContainer] {
            let tapGesture = UITapGestureRecognizer()
            tapGesture.rac_gestureSignal()?.subscribeNext({ [unowned self] (_) -> Void in
                self.linkPressed(view!)
            })
            
            view?.addGestureRecognizer(tapGesture)
        }
        
        if let labels = titleLabels {
            for label in labels {
                label.font = UIFont.detailFont(ofSize: 20.0)
            }
        }
        
        if let labels = bodyLabels {
            for label in labels {
                label.font = UIFont.detailFont(ofSize: 15.0)
            }
        }
    }
    
    @IBAction func linkPressed(sender: UIView) {
        self.openTicket(NSURL(string: "https://www.ticketea.com/entrades-embassat-2015/")!)
    }
    
    func openTicket(link: NSURL) {
        UIApplication.sharedApplication().openURL(link)
    }
}

